lib/record_store/provider/oracle_cloud_dns.rb

```diff
@@ -0,0 +1,195 @@
+require 'oci'
+
+module RecordStore
+  class Provider::OracleCloudDNS < Provider
+    class Error < StandardError; end
+
+    class << self
+      def client
+        config = OCI::Config.new
+        config.user = secrets['user']
+        config.fingerprint = secrets['fingerprint']
+        config.key_content = secrets['key_content']
+        config.tenancy = secrets['tenancy']
+        config.region = secrets['region']
+        config.validate
+        OCI::Dns::DnsClient.new(config: config)
+      end
+
+      # Downloads all the records from the provider.
+      #
+      # Returns: an array of `Record` for each record in the provider's zone
+      def retrieve_current_records(zone:, stdout: $stdout)
+        client.get_zone_records(zone)
+          .flat_map { |response| response.data.items }
+          .flat_map { |api_record| build_from_api(api_record) }
+          .compact
+      rescue StandardError
```

Comment:
Do we really want to catch `StandardError` ? Isn't there a generic `OCI::Error` or such ? Pretty sure we would like to know more about the error if it's not an OCI one. Could also be nice to have the error message as part of the `stdout.puts` below!

Reply:
Absolutely, there's `OCI::Error`. But the thing is it's not categorized as practical Error like `NS1::Error` so I cannot raise those errors in `oracle_cloud_dns`. I can use it on test side for sure. Let me check it more.

Reply:
Maybe catch `OCI::Error` and raises another supported error type that you create ? Like:

```
module RecordStore
  class Provider::OracleCloudDNS
    class Error < SomeGenericErrorTypeForTheRecordStore; end
  end
end
```

Reply:
Ended up using `rescue OCI::Errors::HttpRequestBasedError`